### PR TITLE
Suppress output unless the `git submodule update` call fails

### LINKED ISSUE #2479: git submodule update --init failure needs a better error message
Description of problem:
Pushing a repo that causes `git submodule update --init` to fail gives a generic error message from git that doesn't point to the real problem.  It would be nice to say that there was a problem checking out a submodule.

In my case, this was because I had bundled some dependencies into my repo and the dependency manger had cloned a repo or two into that bundle.

Output of the following commands

- `uname -a`: Linux podcastinet 3.16.0-4-amd64 #1 SMP Debian 3.16.36-1+deb8u2 (2016-10-19) x86_64 GNU/Linux

- `free -m`:
```
             total       used       free     shared    buffers     cached
Mem:          1000        845        155          0         64        584
-/+ buffers/cache:        195        804
Swap:            0          0          0
```
- `docker version`: Docker version 1.12.3, build 6b644ec
- `docker -D info`:
- `sigil -v`:
- `docker run -ti gliderlabs/herokuish:latest herokuish version`:
- `dokku version`: 0.7.2

Environment details (AWS, VirtualBox, physical, etc.): Digial Ocean

How was Dokku installed?: curl | bash

How reproducible: Every time

Steps to Reproduce:
1. Bundle up a common lisp webapp with [fukamachi/qlot](https://github.com/fukamachi/qlot)
2. Save the `quicklisp/` directory to the app repo
3. Attempt to push to dokku

Actual Results: https://gist.github.com/fisxoj/a4f5784e0398d13375cae96a3ee6d906
```
To podcastinet.com:podcastinet
 ! [remote rejected] master -> master (pre-receive hook declined)
error: failed to push some refs to 'dokku@podcastinet.com:podcastinet'
```

Expected Results:

Additional info:

### PR DESCRIPTION
**Pull request by josegonzalez**
This will give a bit more warning to developers as to what might be going on, allowing them to respond appropriately.

Closes #2479

> This was not tested on a real server, so don't merge until I do that.

### CODE REVIEW

**Comment by RealOrangeOne** [comment]
I would say it'd actually be nice to see what's happening when submodules are being fetched. My projects can have submodules that take a long time to sync, and it'd be nice to know what's actually running

File: plugins/git/functions [hunk 1]
```diff
@@ -41,6 +41,21 @@ git_build_app_repo() {
...
+
+  "$@" > "$TMP_COMMAND_OUTPUT" 2>&1 || {
+    local exit_code="$?"
+    cat "$TMP_COMMAND_OUTPUT"
+    return "$exit_code"
+  }
+  return 0
+}
+
+
```
Comment (michaelshobbs) [inline plugins/git/functions]: extraneous newline

File: plugins/git/functions [hunk 2]
```diff
@@ -41,6 +41,21 @@ git_build_app_repo() {
   fi
 }
 
+suppress_output() {
```
Comment (michaelshobbs) [inline plugins/git/functions]: any reason not to place this in common?

File: plugins/git/functions [hunk 3]
```diff
@@ -41,6 +41,21 @@ git_build_app_repo() {
...
+  declare desc="suppress all output from a given command unless there is an error"
+  local TMP_COMMAND_OUTPUT
+  TMP_COMMAND_OUTPUT=$(mktemp "/tmp/${FUNCNAME[0]}.XXXX")
+  trap 'rm -rf "$TMP_COMMAND_OUTPUT" > /dev/null' RETURN INT TERM EXIT
+
+  "$@" > "$TMP_COMMAND_OUTPUT" 2>&1 || {
```
Comment (michaelshobbs) [inline plugins/git/functions]: clever! does this work as expected?

**Comment by michaelshobbs** [comment]
as @RealOrangeOne brings up, perhaps we shouldn't hide the output at all?

File: plugins/git/functions (hunk 2 above)
Comment (josegonzalez) [inline plugins/git/functions]: It's only used here.

File: plugins/git/functions (hunk 3 above)
Comment (josegonzalez) [inline plugins/git/functions]: Yep, though I recommend you add a failing test for me ;)

**Comment by josegonzalez** [comment]
Maybe. I think the submodule output is kinda verbose, so turning it on doesn't have real benefit for most people.

I also think submodules are gross as hell, and you shouldn't ever use them, but thats a separate matter.

File: plugins/git/functions (hunk 2 above)
Comment (michaelshobbs) [inline plugins/git/functions]: Academic argument: it's generic enough that it probably should exist in common. regardless, this sort of thing is an option in the golang `DokkuCmd` struct

File: plugins/git/functions (hunk 3 above)
Comment (michaelshobbs) [inline plugins/git/functions]: I will when this gets ported to golang 😝

File: plugins/git/functions (hunk 2 above)
Comment (josegonzalez) [inline plugins/git/functions]: Cool great. Going forward, outputting all logs to a file and the error output from a failing command to stdout seems reasonable.

**Pull request merged**